invesalius/data/polydata_utils.py

```diff
@@ -50,11 +50,13 @@
 else:
     _has_win32api = False
 
+if TYPE_CHECKING:
+    from vtk import vtkDataObject
 # Update progress value in GUI
 UpdateProgress = vu.ShowProgress()
 
 
-def ApplyDecimationFilter(polydata, reduction_factor):
+def ApplyDecimationFilter(polydata: "vtkDataObject", reduction_factor: float) -> vtkPolyData:
```

Comment:
you can change polydata from `vtkDataObject` to `vtkPolyData` here and in the other functions with polydata